pkg/supervisor/supervisor.go

```diff
@@ -302,5 +302,8 @@ func getEnv(dataDir, component string, keepEnvPrefix bool) []string {
 func (s *Supervisor) GetProcess() *os.Process {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
+	if s.cmd == nil {
```

Comment:
Hmm, is it okay to call GetProcess when the supervisor isn't started?

Reply:
It is now :) prior to checking it the code would panic more times than it didn't.

Also it's thread-safe because s.cmd is only written in: https://github.com/k0sproject/k0s/blob/fa80f8b1fe3f76ae8d67b570a3bff53f3ac25f55/pkg/supervisor/supervisor.go#L168-L178

Reply:
My question was more of the sorts: What is Supervisor's contract? Or what should Supervisor's contract be? Is it a legal thing to call GetProcess before Supervise has been called? There's no go docs about this. I see that we could very well signal the wrong PID in certain circumstances, i.e. the cmd won't be nilled out after the process terminated. Anyhow, that's something that can be tackled another time.

> prior to checking it the code would panic more times than it didn't.

Not sure what you mean.

Reply:
> Not sure what you mean.

Before adding this condition, `s.GetProcess()` could be called before `s.cmd != nil` and this resulted in a panic, of course it was racy, but it happened extremely more than 50% of the times.